evalml/tests/component_tests/test_oversamplers.py

```diff
@@ -341,3 +341,30 @@ def test_oversampler_sampling_dict_strings(oversampler):
 
     assert len(new_X) == sum(expected_result.values())
     assert new_y.value_counts().to_dict() == expected_result
+
+
+@pytest.mark.parametrize("oversampler", [SMOTESampler, SMOTENCSampler, SMOTENSampler])
+@pytest.mark.parametrize(
+    "minority,expected,fails",
+    [(1, 0, True), (2, 1, False), (5, 4, False), (10, 5, False)],
+)
+def test_oversampler_sampling_k_neighbors(minority, expected, fails, oversampler):
+    X = np.array(
+        [
+            [i for i in range(1000)],
+            [i % 7 for i in range(1000)],
+            [0.3 * (i % 3) for i in range(1000)],
+        ]
+    ).T
+    X_ww = infer_feature_types(X, feature_types={0: "Categorical", 1: "Categorical"})
+    y = np.array(["minority"] * minority + ["majority"] * (1000 - minority))
+    overs = oversampler(k_neighbors_default=5)
+    if fails:
+        with pytest.raises(
+            ValueError, match="Minority class needs more than 1 sample to use SMOTE"
+        ):
+            overs.fit_transform(X_ww, y)
+        return
+    _, _ = overs.fit_transform(X_ww, y)
```

Comment:
nit pick, if we're not testing either output then maybe just calling `overs.fit_transform(X_ww, y)` is fine?